### ISSUE TITLE
Global gitignore is not respected.

### ISSUE THREAD

**Issue by ilyagr** [dprint/dprint#855]
Version: 0.46.1

While `dprint` respects my repo's .gitignore ([as advertised](https://github.com/dprint/dprint/blob/51bdc8baaaecf80e19e7ebc77bdc427890773cb9/website/src/config.md?plain=1#L111)), it's still trying to format files in my global ignore file `~/.config/git/ignore` AKA `$XDG_CONFIG_DIR/git/ignore`. It should respect for that file (see also https://git-scm.com/docs/gitignore, in principle it could also support Git's `core.excludesFile` option mentioned in there).

To reproduce, put `_globalignore/*` in `~/.config/git/ignore`, and put some file in `REPO_ROOT/_globalignore/file.md`. `cd REPO_ROOT; dprint fmt` will try to format that file, even though it should not. I use such a dir to keep track of personal files that should not be shared with anybody else working on the repo.

**Comment by adrian-gierakowski** [comment]
If this gets implemented could we please provide a way to disable it? Thank you

**Comment by ilyagr** [comment]
To elaborate on my motivations:

I don't think ignoring the global .gitignore is a very serious problem, but I think it is a problem.

The main reason I think it's a problem is that it means `dprint fmt` might format some non-version-controlled files irreversibly.

The way I noticed it is that I had a very broken file in my `_globalignore` directory, which was meant for scratch documents.

> If this gets implemented could we please provide a way to disable it? Thank you

I'm curious why you'd want that, when would disabling this be convenient? (Note, however, that this is really pure curiosity. I have no authority on how or whether this would be implemented, and no strong opinion on how it should be configured)

**Comment by dsherret** [comment]
Yeah, I think this should be opt-in because it might cause CI failures if a file doesn't get formatted locally due to the global gitignore, but then the CI does format it. Also, it requires shelling out to git to get `git config --get core.excludesFile`, which will slow things down. You can opt-in with `DPRINT_GLOBAL_GITIGNORE=1` in the next version.
